src/Bicep.LangServer.IntegrationTests/HoverTests.cs

```diff
@@ -298,6 +299,59 @@ public async Task Hovers_are_displayed_on_discription_decorator_objects()
                 h => h!.Contents.MarkupContent!.Value.Should().EndWith("```\nthis is my output\n"));
         }
 
+        [DataTestMethod]
+        public async Task Hovers_are_displayed_on_discription_decorator_objects_across_modules()
```

Comment:
Could you also add a tests for description hovers when pointing to ARM JSON files? 

I suspect to get it working, you may need to fetch the description from metadata (if present) in these two places:
https://github.com/Azure/bicep/blob/d286caaddb26cbfdb06541a65f1f6520f23601cd/src/Bicep.Core/Semantics/ArmTemplateSemanticModel.cs#L64-L69
https://github.com/Azure/bicep/blob/d286caaddb26cbfdb06541a65f1f6520f23601cd/src/Bicep.Core/Semantics/ArmTemplateSemanticModel.cs#L81-L84

Note that `TypeProperty` takes an optional description which isn't being set in the above.